microceph/api/disks.go

```diff
@@ -44,3 +56,45 @@ func cmdDisksPost(s *state.State, r *http.Request) response.Response {
 
 	return response.EmptySyncResponse
 }
+
+// cmdDisksDelete is the handler for DELETE /1.0/disks/{osdid}.
+func cmdDisksDelete(s *state.State, r *http.Request) response.Response {
+	var osd string
+	osd, err := url.PathUnescape(mux.Vars(r)["osdid"])
+	if err != nil {
+		return response.InternalError(err)
```

Comment:
Should this be a BadRequest error rather than an InternalError? The consumer of this REST api would know they've provided a malformed request of some sort. It may end up in an InternalError on the client side, etc. But I think the correct response is a BadRequest here.